### ISSUE TITLE
Deck editor - Implement Pauper, Modern, Legacy, Standard, Vintage, EDH legal filters

### ISSUE THREAD

**Issue by spjspj** [magefree/mage#1575]
Basically I'm thinking of a button panel with 'Modern Legal', 'Pauper Legal', 'Commander Legal' etc buttons that will filter the cards being 'Import'ed and 'Load'ed but also act as an 'AND' with the existing search options.

Note there are existing Modern, Standard and Extended filters in the Sets filter drop-down but selecting these still allow banned cards such as Splinter Twin for modern to be found.

**Comment by spacemoses** [comment]
Sounds like it is possibly a straight up bug that these banned cards are showing when a format is selected from the dropdown.  Is a banned card like Felidar Guardian still considered part of standard even though it is banned?  Having 'Legal' versions of the formats could work, but I'd also propose leaving out banned cards by default and having a toggle button to show banned cards OR show the cards in the results and maybe have a different highlighting for the row, like highlighting Felidar Guardian highlighted red if you search for it when having Standard selected from the dropdown.

**Comment by spacemoses** [comment]
I just checked Gatherer and no results are returned when I search for:

Name:
DOES contain "Felidar Guardian"
Format:
DOES contain "Standard"

**Comment by skeate** [comment]
I looked into doing this filtering a bit (specifically, adding a Pauper filter) which is when I discovered that the filters are completely set-based. Hence why Vintage and Legacy are combined.

I don't _think_ it'd be much effort to make these actually filter to legality (at least for the major constructed formats -- block constructed might be harder). It would involve splitting legacy/vintage (and I'd probably add Pauper while I was in there)
